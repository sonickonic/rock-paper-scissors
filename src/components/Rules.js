import React from "react";
import styled from "styled-components";
import IconSrc from "../images/icon-close.svg";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  position: absolute;

  padding: 10rem 0 6rem;
  background-color: var(--white);
  height: 100%;
  width: 100%;
  top: 0;
  right: 0;

  @media (min-width: 500px) {
    padding: 3rem 5rem 3rem;
    height: 41.5rem;
    width: 40rem;
    border-radius: 1rem;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    box-shadow: 0px 0px 0px 100vh rgba(0, 0, 0, 0.7);
  }
`;

const CloseButton = styled.a`
  display: inline-block;
  cursor: pointer;

  @media (min-width: 500px) {
    position: absolute;
    top: 4rem;
    right: 5rem;
  }
`;

const Title = styled.h3`
  color: var(--text);
  font-size: 3.2rem;
  font-weight: 700;

  @media (min-width: 500px) {
    align-self: flex-start;
  }
`;

const Image = styled.img`
  width: 30rem;
`;

const Rules = ({ currentGame, handleClick }) => {
  return (
    <Container>
      <Title>rules</Title>
      <Image alt="Game rules" src={currentGame.rules} />
      <CloseButton>
        <img onClick={handleClick} alt="Close button" src={IconSrc} />
      </CloseButton>
    </Container>
  );
};

export default Rules;
