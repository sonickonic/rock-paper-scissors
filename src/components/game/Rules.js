import React, { useState, useEffect } from "react";
import styled from "styled-components";

import ImageSrc from "../../images/image-rules.svg";
import IconSrc from "../../images/icon-close.svg";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  position: absolute;

  padding: 10rem 0 6rem;
  background-color: var(--white);
  height: 100%;
  width: 100%;
  top: 0;
  right: 0;
`;

const RulesButton = styled.a`
  display: inline-block;
  color: var(--white);
  font-size: 1.6rem;
  letter-spacing: 0.15em;
  border: 1px solid var(--white);
  padding: 1rem 3.6rem;
  border-radius: 0.7rem;
  margin-top: auto;
  cursor: pointer;
`;

const CloseButton = styled.a`
  display: inline-block;
  cursor: pointer;
`;

const Title = styled.h3`
  color: var(--text);
  font-size: 3.2rem;
  font-weight: 700;
`;

function Rules() {
  const [open, setOpen] = useState(false);

  const handleClick = () => {
    open ? setOpen(false) : setOpen(true);
  };

  useEffect(() => {
    const handleEsc = (e) => {
      if (e.keyCode === 27) {
        setOpen(false);
      }
    };
    window.addEventListener("keydown", handleEsc);
  }, []);

  return (
    <div>
      <RulesButton onClick={handleClick}>Rules</RulesButton>
      {open ? (
        <Container>
          <Title>rules</Title>
          <img alt="Game rules" src={ImageSrc} />
          <CloseButton>
            <img onClick={handleClick} alt="Close button" src={IconSrc} />
          </CloseButton>
        </Container>
      ) : null}
    </div>
  );
}

export default Rules;
