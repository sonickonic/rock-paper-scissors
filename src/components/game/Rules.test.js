import React from "react";
import { render, fireEvent } from "@testing-library/react";
import gameType from "../../gameTypes";
import Rules from "./Rules";

describe("Rules", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Rules gameType={gameType} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("calls handleClick prop on close", () => {
    const handleClick = jest.fn();
    const { queryByAltText } = render(
      <Rules gameType={gameType} handleClick={handleClick} />
    );

    fireEvent.click(queryByAltText("Close button"));

    expect(handleClick).toBeCalledTimes(1);
  });
});
