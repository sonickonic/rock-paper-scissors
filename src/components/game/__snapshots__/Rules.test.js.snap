// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rules renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw qAjNL"
  >
    <h3
      class="sc-AxhUy gNFPyL"
    >
      rules
    </h3>
    <img
      alt="Game rules"
      class="sc-AxgMl qlIKz"
    />
    <a
      class="sc-AxhCb hbzvrW"
    >
      <img
        alt="Close button"
        src="icon-close.svg"
      />
    </a>
  </div>
</DocumentFragment>
`;
